target/linux/realtek/files-6.6/drivers/net/dsa/rtl83xx/dsa.c

```diff
@@ -1456,10 +1457,16 @@ static int rtl83xx_vlan_add(struct dsa_switch *ds, int port,
 
 	mutex_lock(&priv->reg_mutex);
 
-	if (vlan->flags & BRIDGE_VLAN_INFO_PVID)
+	if (dsa_port_is_cpu(dp)) {
+		if ((priv->ports[port].pvid == RTL_DEFAULT_VID) || (vlan->vid < priv->ports[port].pvid)) {
+			rtl83xx_vlan_set_pvid(priv, port, vlan->vid);
+			pr_warn("enforce management VLAN %d\n", vlan->vid);
+		}
+	}
+	else if (ispvid)
 		rtl83xx_vlan_set_pvid(priv, port, vlan->vid);
 	else if (priv->ports[port].pvid == vlan->vid)
-		rtl83xx_vlan_set_pvid(priv, port, 0);
+		rtl83xx_vlan_set_pvid(priv, port, RTL_DEFAULT_VID);
```

Comment:
And this.  There should be a way for the user to express "no pvid" on a port.  Allowing untagged frames to enter vlan 1 on a port where that wasn't configured is unexpected, and could be viewed as a security problem.

Reply:
removed from last PR. wont fix.